File: 2017m./uzd1./sprendimai/uzd1.py
hexadecimalValues = ["0", "1", "2", "3", "4",
                     "5", "6", "7", "8", "9",
                     "A", "B", "C", "D", "E", "F"]

def convertToHexadecimal(num):
    hexadecimalValue = num%16
    hexadecimal = hexadecimalValues[hexadecimalValue]
    num -= hexadecimalValue
    hexadecimalValue = int(num/16)
    hexadecimal = hexadecimalValues[hexadecimalValue] + hexadecimal
    return hexadecimal

def main():
    data = open("U1.txt", "r")
    output = open("U1rez.txt", "w")

    size = next(data).split(" ")
    size = [int(i) for i in size]

    xPos = 0

    for i in range(size[0] * size[1]):
        xPos += 1
        rgb = next(data)
        rgb = rgb.split(" ")
        rgb = [int(i) for i in rgb]
        for i in range(3):
            hexadecimal = convertToHexadecimal(rgb[i])
            output.write(hexadecimal)
        if(xPos == size[1]):
            output.write("\n")
            xPos = 0
        else:
            output.write(";")

    data.close()
    output.close()

if __name__ == "__main__":
    main()

File: 2017m./uzd1./sprendimai/uzd1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

using namespace std;

vector<string> hexadecimalValues = {
    "0", "1", "2", "3", "4",
    "5", "6", "7", "8", "9",
    "A", "B", "C", "D", "E", "F"
};

string convertToHexadecimal(int num) {
    int hexadecimalValue = num % 16;
    string hexadecimal = hexadecimalValues[hexadecimalValue];
    num -= hexadecimalValue;
    hexadecimalValue = num / 16;
    hexadecimal = hexadecimalValues[hexadecimalValue] + hexadecimal;
    return hexadecimal;
}

int main() {
    ifstream data("U1.txt");
    ofstream output("U1rez.txt");

    int height, width;
    data >> height >> width;

    int xPos = 0;

    for (int i = 0; i < height * width; i++) {
        xPos += 1;
        vector<int> rgb(3);
        data >> rgb[0] >> rgb[1] >> rgb[2];

        for (int j = 0; j < 3; ++j) {
            string hexadecimal = convertToHexadecimal(rgb[j]);
            output << hexadecimal;
        }

        if (xPos == width) {
            output << "\n";
            xPos = 0;
        }
        else {
            output << ";";
        }
    }

    data.close();
    output.close();

    return 0;
}
